Authentication flow with Firestore

alexdanieldm opened the pull request:
Integrated _Firestore Cloud_ into the authentication flow of the App. And using _Firestore Cloud_, add the following features:
- On `logIn` - READ data for current user
- On `signUp` - add new users to users collection

alexdanieldm committed:
feat(signUp): add new users to firestore cloud

chore(config): remove optional values

style(logOutButton): change font size and padding

duranmla commented on src/firebase/config.js:
```diff
@@ -20,8 +21,8 @@ const firebaseConfig = {
-	appId: APP_ID,
-	measurementId: MEASUREMENT_ID
+	appId: APP_ID
+	// measurementId: MEASUREMENT_ID
```
Why you choose to comment on this instead removing it?

alexdanieldm replied:
Ngl, I was cleaning/testing different config values and forget to delete `measurementId: MEASUREMENT_ID` before pushing the code

duranmla commented on src/screens/signUp.js:
```diff
@@ -5,6 +5,10 @@ import { firebase } from '../firebase/config';
 
 import Loading from '../components/loading';
 
+// ! DEBUG FUNCTIONS ROUTE
```
What are this comments?

alexdanieldm replied:
In this case, that import is only to help me debug (the function log message and states if needed with timestamp so i see better when everything its happening), so the import and all instances of the function should be deleted for the final version and the comment its to help remember that

In the future i'll change it to // ! DELETE AFTER DEBUGGING SESSION

duranmla commented on src/utils/logWithTime.js:
```diff
@@ -15,6 +14,9 @@ const LogWithTime = (message, statesArray) => {
 	}
 	console.log('');
```
Be careful with hanging console logs.
It would be great if you can use https://eslint.org/ _(but take it as a separate thing)_

duranmla reviewed (commented):
It is the feature complete? Or this PR is a draft?

alexdanieldm committed:
fix(config): update firestore settings
* Use experimentalAutoDetectLongPolling to WebChannel transport error

chore: remove unnecessary code

chore(timedLog): rename file previously name "LogWithTime"

alexdanieldm committed:
fix(config): to merge original and custom settings

alexdanieldm committed:
refactor(App): to new authentication flow

duranmla commented on src/App.js:
```diff
@@ -12,36 +12,55 @@ import LogIn from './screens/logIn';
-import LogOutButton from './components/logOutButton'
+import LogOutButton from './components/logOutButton';
+
+// ! DELETE AFTER DEBUGGING SESSION
```
We finish right? Therefore, this comments shouldn't be here.

alexdanieldm replied:
I'll remove it today

duranmla commented on src/App.js:
```diff
@@ -12,36 +12,55 @@ import LogIn from './screens/logIn';
+		firebase.auth().onAuthStateChanged( user => {
 			if (user) {
-				setIsLoggedIn(true);
+				timed_log(`Start Login...`);
```
Be careful with all the `timed_log`

duranmla reviewed (approved):
Make sure to remove debug lines before merging this.

alexdanieldm committed:
perf(package): add commands to "clean" script

chore: remove unnecessary code
